lib/services/template-service.ts

```diff
@@ -83,97 +82,89 @@ export class TemplateService implements ITemplateService {
-
-            //                         })
-            //                         .catch((errorPromises: any) => {
-            //                             // TODO Implement error logger
-            //                             resolve(Backup.fallback);
-            //                         });
-            //                 })
-            //                 .catch((errorTemplates: any) => {
-            //                     console.error(errorTemplates);
-            //                 });
-            //         } else {
-            //             // Load data from cache
-            //             resolve(value);
-            //         }
-            //     }
-            // });
+            tmpCache.get("tempDetails", (error: any, value: any) => {
+                if (!error) {
+                    if (value === undefined) {
+                        this.getTemplatesNames()
+                            .then((templateNames: any) => {
+                                return templateNames;
+                            })
+                            .then((names: any) => {
+                                names.forEach((name: string) => {
+                                    promises.push(
+                                        this.getAppTemplateDetails(name)
+                                            .then((details) => {
+                                                tempDetails.push(details);
+                                            })
+                                            .catch((errorDetails) => {
+                                                reject(errorDetails);
+                                            })
+                                    );
+                                });
+
+                                Promise.all(promises)
+                                    .then(() => {
+                                        tempDetails = this.sortTmpData(tempDetails);
+                                        tmpCache.set("tempDetails", tempDetails, Config.cacheTime);
+                                        resolve(tempDetails);
+
+                                    })
+                                    .catch((errorPromises: any) => {
+                                        // TODO Implement error logger
+                                        resolve(BACKUP.fallback);
+                                    });
+                            })
+                            .catch((errorTemplates: any) => {
+                                console.error(errorTemplates);
+                            });
+                    } else {
+                        // Load data from cache
+                        resolve(value);
+                    }
+                }
+            });
+        });
+    }
+
+    private sortTmpData(templates: Array<any>) {
+        const flavOrder: Array<string> = ["JavaScript", "TypeScript", "Angular & TypeScript"];
```

Comment:
I think we should agree against using this kind of abbreviations -- in my opinion we are not gaining anything by calling this `flavOrder` vs `flavorOrder`